pkg/cmd/extension/command.go

```diff
@@ -39,6 +45,148 @@ func NewCmdExtension(f *cmdutil.Factory) *cobra.Command {
+						query.Sort = sort
+					}
+
+					query.Keywords = args
+					query.Qualifiers = qualifiers
+
+					host, _ := cfg.DefaultHost()
+					searcher := search.NewSearcher(client, host)
+					tp := tableprinter.New(io)
+
+					if webMode {
+						url := searcher.URL(query)
+						if io.IsStdoutTTY() {
+							fmt.Fprintf(io.ErrOut, "Opening %s in your browser.\n", text.DisplayURL(url))
+						}
+						return browser.Browse(url)
+					}
+
+					result, err := searcher.Repositories(query)
+					if err != nil {
+						return err
+					}
+
+					if io.CanPrompt() {
+						fmt.Fprintf(io.Out, "Showing %d of %d gh extensions\n", query.Limit, result.Total)
+						fmt.Fprintln(io.Out)
+						tp.HeaderRow("NAME", "REPO", "INSTALLED", "OFFICIAL", "DESCRIPTION")
+					}
+
+					cs := io.ColorScheme()
+					installedExts := m.List()
+
+					isInstalled := func(repoFullName string) bool {
+						for _, e := range installedExts {
+							// TODO consider a Repo() on Extension interface
+							var installedRepo string
+							if u, err := git.ParseURL(e.URL()); err == nil {
+								if r, err := ghrepo.FromURL(u); err == nil {
+									installedRepo = ghrepo.FullName(r)
+								}
+							}
+							if repoFullName == installedRepo {
+								return true
+							}
+						}
+						return false
+					}
+
+					for _, repo := range result.Items {
+						if !strings.HasPrefix(repo.Name, "gh-") {
+							continue
+						}
+
+						official := ""
+						if repo.Owner.Login == "cli" || repo.Owner.Login == "github" {
+							if io.IsStdoutTTY() {
+								official = "✓"
+							} else {
+								official = "official"
+							}
+						}
+
+						installed := ""
+						if isInstalled(repo.FullName) {
+							if io.IsStdoutTTY() {
+								installed = "✓"
+							} else {
+								installed = "installed"
+							}
+						}
+
+						cmdName := strings.TrimPrefix(repo.Name, "gh-")
+
+						tp.AddField(cmdName, tableprinter.WithColor(cs.Bold))
+						tp.AddField(repo.FullName)
+						tp.AddField(installed, tableprinter.WithColor(cs.Green))
+						tp.AddField(official, tableprinter.WithColor(cs.Yellow))
+						tp.AddField(repo.Description)
+						tp.EndRow()
+					}
+
+					return tp.Render()
+				},
+			}
+
+			// Output flags
+			cmd.Flags().BoolVarP(&webMode, "web", "w", false, "Open the search query in the web browser")
+			cmdutil.AddJSONFlags(cmd, &exporter, search.RepositoryFields)
+
+			// Query parameter flags
+			cmd.Flags().IntVarP(&query.Limit, "limit", "L", 30, "Maximum number of extensions to fetch")
+			cmdutil.StringEnumFlag(cmd, &order, "order", "", "desc", []string{"asc", "desc"}, "Order of repositories returned, ignored unless '--sort' flag is specified")
+			cmdutil.StringEnumFlag(cmd, &sort, "sort", "", "best-match", []string{"forks", "help-wanted-issues", "stars", "updated"}, "Sort fetched repositories")
+
+			// Qualifier flags
+			cmd.Flags().StringVar(&qualifiers.Language, "language", "", "Filter based on the coding language")
```

Comment:
Does it matter in which language is an extension authored in? Would people want to filter on that?

Reply:
my thinking was it might matter for interpreted extensions, but my non-scientific estimate of the current crop of extensions is they are either shell scripts or precompiled. i'm fine taking this out.